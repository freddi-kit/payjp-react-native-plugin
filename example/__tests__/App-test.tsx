/**
 * @format
 */

import "react-native";
import React from "react";
import App from "../src/App";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

jest.mock("payjp-react-native", () => {
    const mockPayjp = {
        PayjpCore: { init: jest.fn() },
        PayjpCardForm: { onCardFormUpdate: jest.fn(), setIOSCardFormStyle: jest.fn() },
        PayjpApplePay: { onApplePayUpdate: jest.fn() }
    };
    return mockPayjp;
});

it("renders correctly", () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
});
