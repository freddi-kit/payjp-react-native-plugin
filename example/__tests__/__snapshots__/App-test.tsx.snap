// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={Object {}}
  >
    <View>
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              }
            }
          >
            Step One
          </Text>
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              }
            }
          >
            Edit 
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              App.tsx
            </Text>
             to replace
             
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              PAYJP_PUBLIC_KEY
            </Text>
             with your PAY.JP public key.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              }
            }
          >
            Card Form
          </Text>
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              }
            }
          >
            Click following button to start card form.
          </Text>
          <View
            style={
              Object {
                "flexDirection": "column",
                "height": 100,
                "justifyContent": "space-evenly",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="start_card_form"
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Add Credit Card（MultiLine）
                </Text>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={Object {}}
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {},
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Add Credit Card（CardDisplay）
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "600",
              }
            }
          >
            Apple Pay
          </Text>
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "400",
                "marginTop": 8,
              }
            }
          >
            Edit 
            <Text
              style={
                Object {
                  "fontWeight": "700",
                }
              }
            >
              APPLE_MERCHANT_ID
            </Text>
             in App.tsx, then start Apple Pay.
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Buy with Apple Pay
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
